src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -161,6 +161,23 @@ internal static bool IsEffectivelyPublicOrInternal(Symbol symbol, out bool isInt
                 case SymbolKind.Discard:
                     return IsSymbolAccessibleCore(((DiscardSymbol)symbol).TypeWithAnnotations.Type, within, null, out failedThroughTypeCheck, compilation, ref useSiteDiagnostics, basesBeingResolved);
 
+                case SymbolKind.FunctionPointer:
+                    var funcPtr = (FunctionPointerTypeSymbol)symbol;
+                    if (!IsSymbolAccessibleCore(funcPtr.Signature.ReturnType, within, throughTypeOpt: null, out failedThroughTypeCheck, compilation, ref useSiteDiagnostics, basesBeingResolved))
+                    {
+                        return false;
```

Comment:
>return false; [](start = 24, length = 13)

Do we report a diagnostics that says which specific type is not accessible?

Reply:
> Do we report a diagnostics that says which specific type is not accessible?

No. But as far as I can tell, there isn't actually a scenario today where this can return false. The only types of symbols allowed in the return type or parameter types are actual TypeSymbols, and we unconditionally return true for all of those types of symbols right below.

---
In reply to: [382344495](https://github.com/dotnet/roslyn/pull/41794#discussion_r382344495) [](ancestors = 382344495)

Reply:
> But as far as I can tell, there isn't actually a scenario today where this can return false. The only types of symbols allowed in the return type or parameter types are actual TypeSymbols, and we unconditionally return true for all of those types of symbols right below.

What about the ```case SymbolKind.NamedType:``` and alike above?

---
In reply to: [384858806](https://github.com/dotnet/roslyn/pull/41794#discussion_r384858806) [](ancestors = 384858806,382344495)

Reply:
True, but we have the same behavior here as generic parameters of NamedTypeSymbols. I'll add a test demonstrating inaccessible error messages.

---
In reply to: [387967547](https://github.com/dotnet/roslyn/pull/41794#discussion_r387967547) [](ancestors = 387967547,384858806,382344495)